public/js/apps/transaction/app.jsx

```diff
@@ -50,6 +51,10 @@ export default class TransactionApp extends Component {
     var state = this.state;
     var SignIn = this.props.SignIn;
     var Transaction = this.props.Transaction;
+    var product = products.get(state.productId);
+    var signInRequired = product.seller.kind !== 'donations';
+    console.log('sign-in required to transact product?', state.productId,
+                signInRequired ? 'Yes': 'No');
```

Comment:
this looks cool, i think we'll need to think about making explicit authentication restrictions in payments-config and not infer it from anything to cope with non-authenticated subscriptions. seperate bug though...